### PR TITLE
New: Add parameter to output formatter output to a file

### LINKED ISSUE #2056: [Feature] Parameter to output formatter output to a file [1.5]
# 🚀 Feature request

## Description

Even though we have a few formatters that output to a directory or a file (excel, html) it will be good to have a parameter or similar so any formatter can output to one. Something like `hint --output results.json`.

Not sure if the CLI shoul try to do something generic, if it should always be done at the formatter level or what. I see pros and cons for each.

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #2056

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

File: packages/hint/src/lib/cli/analyze.ts [hunk 1]
```diff
@@ -361,6 +361,7 @@ export default async (actions: CLIOptions): Promise<boolean> => {
         const formatterOptions: FormatterOptions = {
             config: userConfig || undefined,
             date,
+            output: actions.output,
```
Comment (antross) [inline packages/hint/src/lib/cli/analyze.ts]: How about resolving this to an absolute path against the current working directory here so we don't have to do it in each formatter?

Comment (sarvaje) [reply packages/hint/src/lib/cli/analyze.ts]: done!

File: packages/formatter-json/src/formatter.ts [hunk 2]
```diff
@@ -34,17 +35,30 @@ const debug = d(__filename);
 
 export default class JSONFormatter implements IFormatter {
     /** Format the problems grouped by `resource` name and sorted by line and column number */
-    public format(messages: Problem[]) {
+    public async format(messages: Problem[], target: string | undefined, options: FormatterOptions = {}) {
```
Comment (molant) [inline packages/formatter-json/src/formatter.ts]: @antross @sarvaje what do you think about combining `target` and `options`? `target` is not used by all the `formatters` but `options` yes for the filename output so right now is a bit weird

Comment (sarvaje) [reply packages/formatter-json/src/formatter.ts]: I like the idea

Comment (sarvaje) [reply packages/formatter-json/src/formatter.ts]: @molant done.

**Comment by molant** [comment]
@sarvaje can you please fix the conflicts?

**Comment by sarvaje** [comment]
> @sarvaje can you please fix the conflicts?

@molant done!

**Pull request merged**